### ISSUE TITLE
Order Placed

### ISSUE THREAD

**Issue by jeffkush-sdi** [searchdiscovery/Apollo-Documentation-Connected-Vehicle#12]
[Order Placed](https://github.com/searchdiscovery/Apollo-Documentation-Connected-Vehicle/blob/f92ed977d10722db7c21ff478abfcc4142ee7e72/Data%20Layer%20Events/Order%20Placed.md)

**Comment by jeffkush-sdi** [comment]
@Shocopop 

### 'Order Placed' DataLayer QA

**ISSUE**
- Event needs to be bracketed between a Page Load Started/Completed. 
- `priceTier` parameter is not part of the `price{ }` MOVE to `productInfo{ }`
- Expected parameters: `package_id`, `productDiscount` (may not be applicable?), `variant_id` in the `productInfo[]`; _this use case had (2) these parameters populated for the previous Check Step Encountered event_
-  Expected parameters: `paymentGateway`, `paymentID` in the `payment[]` REMOVE `paymentAmount` parameter from item object (see GIT)
- `tax` value undefined (Discuss?)
- maybe trim to (2) decimals where applicable?

https://cwp.sit.siriusxmguardian.com/manage

![Image](https://github.com/searchdiscovery/Apollo-Documentation-Connected-Vehicle/assets/116372762/91dad7bc-23ce-4aee-839a-6c01f4b293a5)

cc @msawlorSDI

**Comment by Shocopop** [comment]
> * Expected parameters: `paymentGateway`, `paymentID` in the `payment[]` REMOVE `paymentAmount` parameter from item object (see GIT)

The current schema only has `paymentAmount` as a parameter of payment[]

**Comment by jeffkush-sdi** [comment]
> > * Expected parameters: `paymentGateway`, `paymentID` in the `payment[]` REMOVE `paymentAmount` parameter from item object (see GIT)
> 
> The current schema only has `paymentAmount` as a parameter of payment[]
@Shocopop had an incorrect GIT link in top of this PROJ event item; updated now. sry

https://github.com/searchdiscovery/Apollo-Documentation-Connected-Vehicle/blob/f92ed977d10722db7c21ff478abfcc4142ee7e72/Data%20Layer%20Events/Order%20Placed.md

**Comment by Shocopop** [comment]
@jeffkush-sdi This should be fixed now

**Comment by jeffkush-sdi** [comment]
@Shocopop 

ISSUE: With this event we noticed that the previous step to the actual Order Place event firing is incomplete i.e. on the 'review' screen we see a dataLayer push for Page Load Started + Checkout Step Encountered, < but this sequence does not complete with Page Load Completed until the actual Order Placed event is fired > then both the Order Place event push and  the Page Load Completed events are pushed to dataLayer; See screenie _(we really tried to not have this change, but in Adobe Launch this would be a hard work-around and this issue could affect reporting)_

Expected Sequencing

- page load started
- checkout step encountered -- review
- page load completed
------ (Order is Placed)
- page load started
- order placed
- page load completed
------

- Also wanted to verify the value we are now seeing in `paymentGateway` parameter...looks like a unique id not a gateway?
- Ensure IDs in general, are propagating thru the flow with same values/context (we've seen some differences between some steps, but hard to quantify other than they seem different in different steps)
- `productId` was missing from the `productInfo{}`

![Image](https://github.com/searchdiscovery/Apollo-Documentation-Connected-Vehicle/assets/116372762/c47cc203-f272-4624-9412-834eaf285ba5)


(older session)
![Image](https://github.com/searchdiscovery/Apollo-Documentation-Connected-Vehicle/assets/116372762/b987a019-028f-4652-ad80-734177012a02)

cc @msawlorSDI

**Comment by Shocopop** [comment]
@jeffkush-sdi 
Hey Jeff, the `OrderPlaced` event is a bracketed one, and from what I understand we should fire it in between `PageLoadStarted` and `PageLoadCompleted` events.

> Expected Sequencing
> 
> * page load started
> * checkout step encountered -- review
> * page load completed
>   ------ (Order is Placed)
> * page load started
> * order placed
> * page load completed

But you are expecting a standalone event in this case?
Shouldn't we fire multiple events inside a bracket?

PS `productId` is fixed

**Comment by jeffkush-sdi** [comment]
@Shocopop **Yes**,  we went back and forth on this one as to whether we could do it in Adobe Launch, but landed on, we just couldn't ensure tracking stability overall/long-term trying to maintain this with (potential frequently changing flows/steps etc..) in Adobe Launch to convert incoming sequences into the proper hit beacon we really need for Order Placed. We can certainly discuss further, any options that may help you, but yes we would prefer the sequences bucketed as above. cc @msawlorSDI

**Comment by Shocopop** [comment]
@jeffkush-sdi 
Thanks for a quick response!
So to summarize, this even't shouldn't be bracketed, right?

**Comment by jeffkush-sdi** [comment]
@Shocopop NO, we would prefer this event **Order Placed to be bracketed**. I know we originally had discussions on whether or not this 'needed' to be bracketed, but yes, after our internal chat we think this need to stay as bracketed event. 

[https://github.com/searchdiscovery/Apollo-Documentation-Connected-Vehicle/blob/f92ed977d10722db7c21ff478abfcc4142ee7e72/Data%20Layer%20Events/Order%20Placed.md](https://github.com/searchdiscovery/Apollo-Documentation-Connected-Vehicle/blob/f92ed977d10722db7c21ff478abfcc4142ee7e72/Data%20Layer%20Events/Order%20Placed.md)

cc @msawlorSDI

**Comment by Shocopop** [comment]
@jeffkush-sdi Is this sequence no longer valid then?
> Expected Sequencing
> 
> * page load started
> * checkout step encountered -- review
> * page load completed
>   ------ (Order is Placed)
> * page load started
> * order placed
> * page load completed

Should it be like that:
* page load started
* checkout step encountered -- review
* page load completed
* page load started
* order placed
* page load completed
* page load started
* order placed
* page load completed

Or should I leave it the same way that it is right now? (Which makes sense because both checkout step(review) and order placed belong to the same virtual page)
* page load started
* checkout step encountered -- review
* order placed
* page load completed
* page load started
* order placed
* page load completed

**Comment by jeffkush-sdi** [comment]
@Shocopop Want to get @msawlorSDI input on this one. keep you posted soon, thx.

**Comment by jeffkush-sdi** [comment]
**(STAGE) ORDER COMPLETE SEQ. Know we're in progress with this sequence...**


![Image](https://github.com/searchdiscovery/Apollo-Documentation-Connected-Vehicle/assets/116372762/4ab1ec49-bfb0-41ef-8013-61e877c35541)

**Comment by jeffkush-sdi** [comment]
@Shocopop 
(STAGE)
Think we know about these from .sit QA, these from STG.

**ISSUE:** Are the IDs in productInfo changing as the flow moves to Order Placed. The IDs do appear to be different/inconsistent and/or maybe being utilized in different context for SXM? Discuss? NOTE `package_id`, `productDiscount`, missing `productID`, `paymentGateway` value

(same session)
Add to Cart

![Image](https://github.com/searchdiscovery/Apollo-Documentation-Connected-Vehicle/assets/116372762/57abb35a-ade5-4af6-b5c5-28a71b1d9d27)

Order Placed


![Image](https://github.com/searchdiscovery/Apollo-Documentation-Connected-Vehicle/assets/116372762/023a60f6-be7f-4943-b1df-cf32c94e2839)


cc @msawlorSDI

**Comment by Shocopop** [comment]
@jeffkush-sdi 
This should be fixed now, I have tested it.
Payment Gateway is now working as requested as well.

**Comment by jeffkush-sdi** [comment]
@Shocopop NICE work...will run by @msawlorSDI on MON.

**QA .SIT**

**NO ISSUE** (minor) `tax` _value was not applied, but have not seen this is any previous QA..._

The dataLayer sequence looks correct from a UX POV and as requested. All parameter IDs appear to be consistent with previous steps. _We know the user sequences on .STG may be different so when we spot QA .STG this is something we'll be looking @ step values in sync w/pageName, product ids > steps, bracketing, order placed etc..._

![Image](https://github.com/searchdiscovery/Apollo-Documentation-Connected-Vehicle/assets/116372762/fe092bff-6866-4a57-8366-3464f8772133)
